pkg/client/client.go

```diff
@@ -196,16 +202,46 @@ func GetCredentialRefForCluster(nutanixCluster *infrav1.NutanixCluster) (*creden
 	if nutanixCluster == nil {
 		return nil, fmt.Errorf("cannot get credential reference if nutanix cluster object is nil")
 	}
-	prismCentralinfo := nutanixCluster.Spec.PrismCentral
-	if prismCentralinfo == nil {
+	prismCentralInfo := nutanixCluster.Spec.PrismCentral
+	if prismCentralInfo == nil {
 		return nil, nil
 	}
-	if prismCentralinfo.CredentialRef == nil {
+	if prismCentralInfo.CredentialRef == nil {
 		return nil, fmt.Errorf("credentialRef must be set on prismCentral attribute for cluster %s in namespace %s", nutanixCluster.Name, nutanixCluster.Namespace)
 	}
-	if prismCentralinfo.CredentialRef.Kind != credentialTypes.SecretKind {
+	if prismCentralInfo.CredentialRef.Kind != credentialTypes.SecretKind {
 		return nil, nil
 	}
 
-	return prismCentralinfo.CredentialRef, nil
+	return prismCentralInfo.CredentialRef, nil
+}
+
+func validateAndSanitizePrismCentralInfoAddress(address string) (string, error) {
```

Comment:
@faiq what are your thoughts on doing all of this in validating/mutating webhooks? I think it would really clean up this file and simplify these functions.

Reply:
there's been a lot of push back against doing this with validating webhooks.  #356 addresses the validation using CEL

Reply:
Even better 🙏